Add `@oneOf` support to introspection query

maciesielka opened the pull request:
Fixes #4077

maciesielka commented on src/utilities/getIntrospectionQuery.ts:
```diff
@@ -32,6 +32,10 @@ export interface IntrospectionOptions {
+
+  // Whether target GraphQL server supports `@oneOf` input objects.
+  // Default: false
+  inputObjectOneOf?: boolean;
```
Followed the pattern here above for making this configurable, but wasn't sure if that was necessary.

mmgoodnow commented on src/utilities/getIntrospectionQuery.ts:
```diff
@@ -32,6 +32,10 @@ export interface IntrospectionOptions {
    * Default: false
    */
   inputValueDeprecation?: boolean;
+
+  // Whether target GraphQL server supports `@oneOf` input objects.
+  // Default: false
```
should this be JSDoc formatted to match the comments above?

maciesielka replied:
🤦🏻 definitely. updated in [3af4434](https://github.com/graphql/graphql-js/pull/4078/commits/3af443467cdb9713f700e2b7b4368fb55c18171c)

mmgoodnow commented on src/utilities/getIntrospectionQuery.ts:
```diff
@@ -223,6 +230,7 @@ export interface IntrospectionObjectType {
   readonly interfaces: ReadonlyArray<
     IntrospectionNamedTypeRef<IntrospectionInterfaceType>
   >;
+  readonly isOneOf: boolean;
```
looks like this was added to the `IntrospectionObjectType` here in addition to the `IntrospectionInputObjectType` below—is that intentional? The other references in this diff seem to only care about input objects.

maciesielka replied:
You're so right. Removed in [633331a](https://github.com/graphql/graphql-js/pull/4078/commits/633331aded549b3db00296646ad4833b5f8b5c7f)

maciesielka commented:
@JoviDeCroock, @mmgoodnow appreciate your reviews and hopefully I've addressed your feedback. Any idea what it takes to get something like this merged?

benjie reviewed (approved):
Thanks for adding this vital feature!
